rm tag default anim, add animate demo

jljsj33 opened the pull request.

zombieJ committed:
Merge pull request #14596 from ant-design/feature-merge-master

Feature merge master

zombieJ committed:
update style

zombieJ committed:
Merge pull request #14614 from ant-design/rc-select

update rc-select to fix empty in short select

Commit:
Feat timepicker clearicon (#14556)

* :sparkles: custom clearIcon for time-picker

* :memo: update readme

ycjcl868 committed:
Merge master into merge-master

zombieJ committed:
Merge pull request #14726 from ant-design/merge-master

Merge master

vinhnph-ibl committed:
Table column filter could be ReactNode

zombieJ committed:
merge master

zombieJ committed:
Merge pull request #14792 from ant-design/feature-merge-master

Feature merge master

zombieJ committed:
Countdown support `onFinish` (#14791)

* support onFinish

* clean up

* update test case

afc163 committed:
🌟 support extra for Collapse.Panel

close #11626

vinhnph-ibl committed:
Merge branch 'feature' into ts_table

vinhnph-ibl committed:
Merge branch 'feature' into ts_table

afc163 committed:
:zap: Use semantic tag names in Layout (#14806)

close #14801

vinhnph-ibl committed:
Merge branch 'feature' into ts_table

afc163 committed:
🌟 support goto button in Pagination (#14819)

close #14798
close #14535
close #10201
close #7328

vinhnph-ibl committed:
Merge branch 'ts_table' of github.com:vinhnph-ibl/ant-design into ts_table

vinhnph-ibl committed:
Merge remote-tracking branch 'upstream/feature' into ts_table

vinhnph-ibl committed:
ReactNode has already included string

yesmeck committed:
Merge pull request #14777 from vinhnph-ibl/ts_table

Table column filter could be ReactNode

zombieJ committed:
merge master

zombieJ committed:
Merge pull request #14883 from ant-design/merge-master

Merge master

zombieJ committed:
New Component: Typography (#14250)

* text with prefix

* add edit style

* support editable

* enhance accessibility & type experience

* optimize IME case

* support copy

* add locale

* add secondary & disabled

* add ellipsis shadow text

* split to 3 components

* update snapshot

* update desc

* change lines also need update ellipsis

* skip aria when is in ellipsis

* add ResizeObserver in _util

* update snapshot

* move TestBase into test file

* update test case

* update doc

* fix typo

* important => level

* use rows

* update demo cols to 1

* fix cssText not work in firefox

* update doc

* add miss point

* support extendable

* update snapshot

* fix doc

* copyable support string

* update snapshot

* update doc

* update doc desc

* adjust style

* full test

* reset after test

* rename

* update snapshot

* fix compile

* adjust style

* update desc

* update prefixCls

* update margin

* adjust

* nest wrap of tag content

* adjust style

* update comment

* rm %

* one more thing

* tmp of measure

* merge string as children

* update snapshot

* update testcase

* remove comment

* use internal variable for configProvider passing

* update snapshot

* use expandable instead of extendable

* less variable it

* update demo

* update less

* adjust code & mark style

* remove mark padding

* update measure logic

* support nest element style

* use childNode.textContent to fix react 15 error

* update css

* popout Typography

* add link style

* adjust doc

* use ellipsis instead of rows & expandable

* update doc

* update doc

* update doc & style

* fix typo

* add css ellipsis support

* client render

* update snapshot

* enhance copyable

* support onExpand

* update test case

* add test of css ellipsis

* fix logic in react 15

* rename onChange -> onSave

* use tagName of article

* fix lint

zombieJ committed:
search input sticky

afc163 committed:
:up: Upgrade rc-input-number@4.4.0 to fix 13896

zombieJ committed:
Form support `labelCol` & `wrapperCol` props. (#15038)

* use context

* update doc

* update doc

* not pass context if nest

* update snapshot

* update doc

* update doc

zombieJ committed:
add icons (#15047)

jljsj33 committed:
rm tag default anim, add animate demo

jljsj33 committed:
rm tag default anim, add animate demo

afc163 commented on components/tag/demo/animation.md:
```diff
@@ -0,0 +1,172 @@
+          </h3>
+          <Animate component="div" transitionName="antd-demo-tag-zoom">
+            {tagChild}
+          </Animate>
```
为啥要同时用到 Animate 和 TweenOne？理论上用一个不就够了。

jljsj33 replied:
有选择性。。。哈哈，，，Animate 是用 max-width，，，有点慢半拍，，所以加了 tween-one 的。，，，辅一下也没事。。。

afc163 replied:
这样实现太恶心了，demo 代码也要优雅一点。

afc163 commented on components/tag/index.tsx:
```diff
@@ -119,28 +122,19 @@ class Tag extends React.Component<TagProps, TagState> {
-          transitionName={`${prefixCls}-zoom`}
-          onEnd={this.animationEnd}
+        <InnerTag
+          show={visible}
```
不需要 show 了。

afc163 replied:
InnerTag 也不用了。

jljsj33 committed:
rm InnerTag

jljsj33 committed:
rm InnerTag

jljsj33 committed:
rm animate

jljsj33 committed:
rm animate

jljsj33 committed:
rm afterClose test

jljsj33 committed:
rm afterClose test

afc163 commented on components/tag/index.en-US.md:
```diff
@@ -18,7 +18,7 @@ Tag for categorizing or markup.
 | Property | Description | Type | Default |
 | -------- | ----------- | ---- | ------- |
-| afterClose | Callback executed when close animation is completed | () => void | - |
+| afterClose | Callback executed when close animation is completed, please use `onClose`, we will delete this in the next version | () => void | - |
```
delete => removed

afc163 commented on components/tag/index.tsx:
```diff
@@ -53,11 +43,24 @@ class Tag extends React.Component<TagProps, TagState> {
+    if (props.afterClose) {
+      console.warn(
+        "Warning: 'afterClose' will be discarded, please use 'onClose', we will delete this in the next version.",
+      );
```
参考其他组件用 warning 函数。

discarded => deprecated

afc163 commented on components/tag/demo/animation.md:
```diff
@@ -0,0 +1,130 @@
+  }
+
+  handleInputConfirm = () => {
+    const state = this.state;
```
state 变量是多余的。

afc163 commented on components/tag/demo/animation.md:
```diff
@@ -0,0 +1,130 @@
+        ) : tagElem}
+      </span>
+    );
+  }
```
这里的逻辑和动画关系不大，可以清理一下。

jljsj33 committed:
update warning

jljsj33 committed:
update warning

jljsj33 committed:
update animation demo

jljsj33 committed:
update animation demo

ycjcl868 committed:
:lipstick: fix dropdown trigger align type def

ztplz commented on components/tag/index.en-US.md:
```diff
@@ -18,7 +18,7 @@ Tag for categorizing or markup.
 | Property | Description | Type | Default |
 | -------- | ----------- | ---- | ------- |
-| afterClose | Callback executed when close animation is completed, please use `onClose`, we will delete this in the next version | () => void | - |
+| afterClose | Callback executed when close animation is completed, please use `onClose`, we will removed this in the next version | () => void | - |
```
`removed` => `remove`,  表示现在的意愿后面接一般现在时

ztplz commented on components/tag/index.tsx:
```diff
@@ -45,11 +46,10 @@ class Tag extends React.Component<TagProps, TagState> {
-    }
+    warning(
+      !('afterClose' in props),
+      "'afterClose' will be deprecated, please use 'onClose', we will removed this in the next version.",
```
同上

afc163 replied:
这里是对的，被动语态

ztplz replied:
![image](https://user-images.githubusercontent.com/17755202/53471512-ce7cb080-3a9f-11e9-91aa-c7f802cb93e9.png)
是这个

zombieJ committed:
skip

zombieJ committed:
adjust

zombieJ committed:
add component param in warning (#15078)

* add component param in warning

* update snapshot

zombieJ committed:
canel when page change (#15082)

zombieJ committed:
fix package.json

zombieJ committed:
fix rc-select version

afc163 committed:
:up: upgrade stylelint-config-prettier to 5.0.0

zombieJ committed:
update doc about customize form item component (#15088)

* update doc

* rm test code

* update doc

afc163 committed:
:bug: fix upload instance cannot exposed in TypeScript (#15090)

close #14439

jljsj33 committed:
fix typo

jljsj33 committed:
fix typo

zombieJ committed:
merge master

zombieJ committed:
Merge pull request #15097 from ant-design/merge-master

Merge master

vitaliymaz committed:
add responsive property to carousel (#15071)

* add responsive property to carousel #15065

* extend carousel props from Slick type definitions #15065

zombieJ committed:
rm key in OptionProps (#15104)

zombieJ committed:
Select multiple mode support showArrow (#15091)

* support showArrow

* fix rc-select version

* update snapshot

zombieJ committed:
update snapshot (#15108)

zombieJ committed:
RangePicker support customize separator (#15055)

* support separator

* add version

* fix css

zombieJ committed:
update doc

yoyo837 committed:
Drawer Body padding should be variable

zombieJ committed:
fix cascader should tab twice to exist (#15117)

* fix cascader should tab twice to exist

* update snapshot

DiamondYuan committed:
up

onlyann committed:
fix DatePicker to support an array of formats (#15116)

ref #13148

DiamondYuan committed:
up

xiaohuoni committed:
table reader mobile friendly

chenshuai2144 committed:
New Component PageHeader (#13637)

* add api dock

* add a simple demo

* add content demos

* fix style warning

* Replace content with children

* add all demo

* add demo test

* add backNode props

* Adjust the divider style

* fix icon do no show

* fix test ci

* fix md code style

* remove fragment use

* add test script

* Missing newline character at end of file

* optimize the demo display

* text becomes longer

* text becomes longer

* fix css code style

* fix test snapshot

* fix css code style

* fix codereivew waring

* fix codereivew waring

* update snapshots

* use ant-page-header-have-footer

* use englist

* fix typo

* updated snapshot

* line-height set to 100%

* remove subtitle

* Specification less

* fix less style error

* Sort by alphabet

* Rebuild code

* use english

* number use Statistic

* snapshot updated

* rm onBack in breadcrumb

* remove unuse return

* add a click wave

* remove unuse css

* add icon hover

* newline character

* new description

* warm descriptive text

* add dot

* remove async

* feat: use Typography.Paragraph

* test: snapshots updated

* style: fix  style lint

* import typography style

afc163 committed:
:bug: Fix closable Drawer hiding without transition

afc163 committed:
:memo: make pull request template more practical

Frezc committed:
add export to MenuItemProps

orzyyyy committed:
test: update cases for calculateNodeHeight

orzyyyy committed:
test: update cases for TextArea

chenshuai2144 committed:
3.14.0 Changelog (#15145)

* add responsive property to carousel (#15071)

* add responsive property to carousel #15065

* extend carousel props from Slick type definitions #15065

* rm key in OptionProps (#15104)

* update doc

* fix cascader should tab twice to exist (#15117)

* fix cascader should tab twice to exist

* update snapshot

* Drawer Body padding should be variable

* fix DatePicker to support an array of formats (#15116)

ref #13148

* table reader mobile friendly

* up

* up

* doc: add 3.14.0 changelog

* doc: fix ci warning

* doc: fix ci warning

* doc: fix typo

* doc: fix ci warning

* doc: fix ci warning

* doc: fix cr warring

* doc: fix cr warring

* doc: fix cr warring

* doc: fix cr warring

* doc: fix cr warring

* doc: fix cr warring

* doc: fix cr warring

* -m

* -m

* doc: fix cr warning

chenshuai2144 committed:
version: up to 3.14.0

orzyyyy committed:
test: update aria-label for TextArea, to fix accesslint problems

orzyyyy committed:
revert: test command in package.json

chenshuai2144 committed:
Merge branch 'master' into master-merge-feature

chenshuai2144 committed:
Merge pull request #15155 from ant-design/master-merge-feature

Master merge feature

chenshuai2144 committed:
Merge pull request #15154 from ant-design/feature

up to 3.14.0

zombieJ committed:
fix configProvider getPopupContainer not work with DatePicker

orzyyyy committed:
test: upadte cases for Transfer.List

orzyyyy committed:
test: update cases for Grid.Row

zombieJ committed:
fix form warning (#15160)

* fix form warning

* update snapshot

chenshuai2144 committed:
feat: PageHeader support classname (#15159)

Hazyzh committed:
:bug: (Site_Icon_Display) modify background color for two tone icon item

when theme equal twoTone add calss to li tag, modify hover color from primary color to #8ecafe.

Closes #1

liuchuzhang committed:
fix document

chenshuai2144 committed:
doc: supplement changelog

afc163 committed:
Merge pull request #15172 from ant-design/changelog

supplement  3.14.0 Changelog

afc163 committed:
Merge pull request #15169 from liuchuzhang/fix-document

fix #15164

chenshuai2144 committed:
doc: add 3.14.1 changlog

chenshuai2144 committed:
doc: replace link to preview

chenshuai2144 committed:
version: up to 3.14.1

afc163 committed:
Merge pull request #15156 from ant-design/date-picker-popup

fix configProvider getPopupContainer not work with DatePicker

afc163 committed:
Merge pull request #15148 from ant-design/improve-pull-request-template

Opitimize the pull request template

afc163 committed:
:bug: Fix Collapse extra node style missing

close #15174

chenshuai2144 committed:
doc: supplement changelog

chenshuai2144 committed:
Merge pull request #15176 from ant-design/fix-collapse-extra

Fix Collapse extra node style missing

chenshuai2144 committed:
Merge pull request #15175 from ant-design/changelog

Add 3.14.1 changelog

chenshuai2144 committed:
doc: fix md style error

Hazyzh committed:
add symbol , in end

chenshuai2144 committed:
doc: fix md style error

afc163 committed:
:art: Improve code style of Input.Password

afc163 committed:
:bug: Fix Input.Password lost focused state issue

close #15173

afc163 committed:
:memo: Opatimize Input demo

afc163 committed:
:bug: Focus input after clear input content

afc163 committed:
Merge pull request #15153 from zy410419243/issue-14775

pref: lift up coverage rate

afc163 committed:
Merge pull request #15170 from Hazyzh/modify_doc_icon_bg

 (Site_Icon_Display) modify background color for two tone icon item

afc163 committed:
Update PULL_REQUEST_TEMPLATE.md

afc163 committed:
:white_check_mark: Add test cases

afc163 committed:
Merge pull request #15184 from ant-design/fix-input-focus-state

Fix Input Password focus state

afc163 committed:
:memo: Remove `--save` after npm install

zombieJ committed:
update ts def

afc163 committed:
Merge pull request #15197 from ant-design/typography-ts

update `setContentRef` ts def

ngolin committed:
Fix: TypeScript <Layout> requires a `tagName` property (#15181)

* fix #15162

* force safe type casting

* get rid of non-null assertion

* get rid of non-null assertion

afc163 committed:
:memo: Fix markdown style

Frezc committed:
[doc]remove misleading content

afc163 committed:
:bug: Text ellipsis should be optional, close #15189

afc163 commented:
ci broken

afc163 committed:
Merge pull request #15209 from ant-design/fix-ellipsis-ts

Text ellipsis should be optional

afc163 committed:
:lipstick: unified DatePicker and TimePicker cursor style

afc163 committed:
Merge pull request #15218 from ant-design/unified-date-picker-cursor-style

Unified DatePicker and TimePicker cursor style

afc163 committed:
Merge pull request #15219 from Frezc/master

[doc]remove misleading content

afc163 committed:
:art: Fix eslint warning

afc163 committed:
:lipstick: prettier codes

afc163 committed:
Merge pull request #15222 from ant-design/prettier-code

Prettier codes

jljsj33 commented:
这飞机是啥？？？codecov 减了0.06 没办法的，，，代码少了。。。

jljsj33 commented:
wraning 改了？？

```js
 warning(
      !('afterClose' in props),
      "'afterClose' will be deprecated, please use 'onClose', we will remove this in the next version.",
    );
``
```
error TS2554: Expected 3 arguments, but got 2.
```

jljsj33 committed:
update wraning

jljsj33 committed:
Merge branch 'rm-tag-anim' of github.com:ant-design/ant-design into rm-tag-anim
